frontend/public/components/environment.jsx

```diff
@@ -21,16 +22,31 @@ const NameValueEditorComponent = (props) => <AsyncComponent loader={() => import
  * @private
  */
 const getPairsFromObject = (element) => {
+  let returnedPairs = {};
   if (_.isUndefined(element.env)) {
-    return [['', '', 0]];
+    returnedPairs.env = [['', '', 0]];
+  } else {
+    returnedPairs.env = _.map(element.env, (leafNode, i) => {
+      if (!_.has(leafNode, 'value') && !_.has(leafNode, 'valueFrom')) {
+        leafNode.value = '';
+      }
+      leafNode.ID = i;
+      return Object.values(leafNode);
+    });
   }
-  return _.map(element.env, (leafNode, i) => {
-    if (!_.has(leafNode, 'value') && !_.has(leafNode, 'valueFrom')) {
-      leafNode.value = '';
-    }
-    leafNode.ID = i;
-    return Object.values(leafNode);
-  });
+  if (_.isUndefined(element.envFrom)) {
+    const configMapSecretRef = {name: '', key: ''};
+    returnedPairs.envFrom = [['', {configMapSecretRef}, 0]];
+  } else {
+    returnedPairs.envFrom = _.map(element.envFrom, (leafNode, i) => {
+      if (!_.has(leafNode, 'prefix')) {
+        leafNode.prefix = '';
+      }
+      leafNode.ID = i;
```

Comment:
We shouldn't need to set the props on the object if we're removing them with _.pick later anyway.

```js
const prefix = leafNode.prefix || '';
const refs = _.pick(leafNode, ['configMapRef', 'secretRef']);
return [prefix, refs, i];
```

Reply:
We are not setting these props on the object, as its passed in as the full object with all its properties. Here we are picking weather configMapRef or secretRef to pass along to the editor.

Reply:
But we're modifying `leafNode`, which is changing `element.envFrom`, correct?